File: ips/ips-model/src/main/java/io/iochord/apps/ips/data/services/models/ImportCsvConfiguration.java
package io.iochord.apps.ips.data.services.models;

import java.io.InputStreamReader;

import com.fasterxml.jackson.annotation.JsonIgnore;

import lombok.Getter;
import lombok.Setter;

/**
 *
 * @package ips-simulator-web
 * @author  Iq Reviessay Pulshashi <pulshashi@ideas.web.id>
 * @since   2019
 *
 *
 */
public class ImportCsvConfiguration {
	
	@Getter
	@Setter
	private String title = "";
	
	@Getter
	@Setter
	private char delimeter = '|';
	
	@Getter
	@Setter
	private int headerRowIdx = 0;
	
	@Getter
	@Setter
	private String filename;
	
	@Getter
	@Setter
	@JsonIgnore
	private InputStreamReader reader;
	
}

File: ips/ips-simulator-web/src/main/ts/src/iochord/ips/common/service/data/models/ImportCsvConfiguration.ts
/**
 * Parameter settings for importing CSV file.
 *
 * @export
 * @class ImportCsvConfiguration
 *
 * @package ts
 * @author I. R. Pulshashi <pulshashi@ideas.web.id>
 * @since 2019
 */
export default class ImportCsvConfiguration {

  /**
   * The name of CSV file
   *
   * @type {string}
   * @memberof ImportCsvConfiguration
   */
  public name: string = '';

  /**
   * Delimiter of the csv file.
   *
   * @type {string}
   * @memberof ImportCsvConfiguration
   */
  public delimeter: string = ';';

  /**
   * The index of header row.
   *
   * @type {number}
   * @memberof ImportCsvConfiguration
   */
  public headerRowIdx: number = 0;
}
